scripts/convert_hf_checkpoint.py

```diff
@@ -41,53 +58,81 @@ def permute(w):
-                sd_hf[f"model.layers.{i}.post_attention_layernorm.weight"]
-            )
+            print("Processing block %s" % i)
+
+            for layer_name, bin_file in bin_index["weight_map"].items():
+                if f"model.layers.{i}." in layer_name:
+                    # Load the weight tensor from the .bin file,
+
+                    hf_weights = load_weights_from_bin(hf_checkpoint_path, bin_file)
+                    print(f"Loading layer {i} weights from bin file {bin_file}")
+                    sd[f"transformer.h.{i}.attn.c_proj.weight"].copy_(
+                        hf_weights[f"model.layers.{i}.self_attn.o_proj.weight"]
+                    )
+
+                    sd[f"transformer.h.{i}.attn.c_attn.weight"][:qkv_size] = permute(
+                        hf_weights[f"model.layers.{i}.self_attn.q_proj.weight"]
+                    )
+                    sd[f"transformer.h.{i}.attn.c_attn.weight"][qkv_size:-qkv_size] = permute(
+                        hf_weights[f"model.layers.{i}.self_attn.k_proj.weight"]
+                    )
+                    sd[f"transformer.h.{i}.attn.c_attn.weight"][-qkv_size:] = hf_weights[
+                        f"model.layers.{i}.self_attn.v_proj.weight"
+                    ]
+
+                    sd[f"transformer.h.{i}.mlp.c_fc1.weight"].copy_(
+                        hf_weights[f"model.layers.{i}.mlp.gate_proj.weight"]
+                    )
+                    sd[f"transformer.h.{i}.mlp.c_fc2.weight"].copy_(
+                        hf_weights[f"model.layers.{i}.mlp.up_proj.weight"]
+                    )
+                    sd[f"transformer.h.{i}.mlp.c_proj.weight"].copy_(
+                        hf_weights[f"model.layers.{i}.mlp.down_proj.weight"]
+                    )
+
+                    sd[f"transformer.h.{i}.rms_1.scale"].copy_(
+                        hf_weights[f"model.layers.{i}.input_layernorm.weight"]
+                    )
+                    sd[f"transformer.h.{i}.rms_2.scale"].copy_(
+                        hf_weights[f"model.layers.{i}.post_attention_layernorm.weight"]
+                    )
+
+                    # Load globals, could be done in a cleaner way. It assumes that the globals will happen to be in a .bin
+                    # ... file that also contains some layers, hopefully always the case
+                    if "model.embed_tokens.weight" in hf_weights:
+                        sd["transformer.wte.weight"].copy_(hf_weights["model.embed_tokens.weight"])
+
+                    if "model.norm.weight" in hf_weights:
+                        sd["transformer.ln_f.scale"].copy_(hf_weights["model.norm.weight"])
+
+                    if "lm_head.weight" in hf_weights:
+                        sd["lm_head.weight"].copy_(hf_weights["lm_head.weight"])
+
+                    # Break here to avoid reloading the same weights multiple times
+                    # this assumes the layers won't be split between bin files, which hopefully is not possible
```

Comment:
the original llama weights are indeed split between the files (because it was tensor-parallel), but it is not the case with HF models AFAIK

Reply:
got it, so we should be ok with this assumption then?